components/brave_new_tab_ui/api/stats.ts

```diff
@@ -20,12 +20,37 @@ export type Stats = {
   bandwidthSavedStat: number
 }
 
+// This is a 'good enough' (for now) solution to a problem which occurred in the
+// base::Value migration. These are uint64s in C++ land, and previously these
+// were being serialized as JS numbers (which have a max value of 2^53 - 1). The
+// new serialization converts them (correctly) to strings. Unfortunately, this
+// breaks our localization logic, which just `.toLocaleString()`s the numbers.
+// A better solution would be to migrate these to all `BigInt`s. However,
+// currently the version of Typescript we're targeting doesn't support BigInt
+// literals, and the change required is, unfortunately, somewhat infectious.
```

Comment:
Could you elaborate on the "somewhat infectious" part? If BigInt is more desirable, but literals are out of the question, why not use the BigInt constructors?

Reply:
I did actually start by doing that but it makes the code pretty ugly, and you end up needing to use them all over the place. I actually had a bit of a think after writing this comment, and for this use case, I think losing precision is actually fine, because for the most part, these numbers are estimates anyway :smile: